coffee-shop-application/src/main/kotlin/com/motycka/edu/order/OrderRepositoryImpl.kt

```diff
@@ -0,0 +1,61 @@
+package com.motycka.edu.order
+
+import org.jetbrains.exposed.sql.Database
+import org.jetbrains.exposed.sql.transactions.transaction
+
+class OrderRepositoryImpl(private val database: Database,
+                          private val orderItemRepository: OrderItemRepository) : OrderRepository {
+
+    override fun selectAll(): List<OrderDTO> = transaction(database) {
+        OrderEntity.all().map { orderEntity ->
+            val orderItems = orderItemRepository.selectByOrderId(orderEntity.id.value)
```

Comment:
It is sometimes a tough decision to decide whether implement such logic in repository or service. In my solution, I chose to do it in service because I felt like it is business logic. But having it in repository is not wrong either, it depends.